### PR TITLE
mgr: fix bugs in init, beacons

### PR DESCRIPTION
**Pull request by liewegas**

### CODE REVIEW

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-04-06_17:19:30-rados:thrash-wip-mgr-init---basic-smithi/

the pg dump failures are gone!  (and we have a new set of regressions.. but they are in master and unrelated to mgr)

**Comment by liewegas** [comment]
retest this please

File: src/mon/MgrMonitor.cc [hunk 1]
```diff
@@ -159,6 +159,10 @@ bool MgrMonitor::preprocess_beacon(MonOpRequestRef op)
     return true;
   }
 
+  if (!mon->is_leader()) {
```
Comment (tchaikov) [inline src/mon/MgrMonitor.cc]: maybe a better fix is to just do the minimum check in `MgrMonitor::preprocess_beacon()`: just check the cap and fsid, and move the other checks to `prepare_beacon()`. please note that we update `last_beacon[]` twice, which does not make sense.

**Comment by liewegas** [comment]
updated with simpler fix

**Pull request merged**